src/App.tsx

```diff
@@ -1,24 +1,122 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { FilterType } from './types/filter';
+import { Main } from './components/Main';
+import { Footer } from './components/Footer';
+import { USER_ID, ERROR_TIMEOUT, FETCH_ERROR } from './utils/constants';
+import { Header } from './components/Header';
 
-const USER_ID = 0;
+const filterActive = (
+  data: Todo[],
+) => data.filter((todo) => todo.completed === false);
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [loadingTodos, setLoadingTodos] = useState<Todo[]>([]);
+
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
```

Comment:
looks like redundant state isLoading. you can use tempTodo state for it.
if you need boolean value - just set `!!tempTodo`